chore: Change app id, update Firebase config, etc.

BumbleFlash opened the pull request:
Changed app ID to edu.usf.sas.pal.muser

BumbleFlash committed:
changed app ID

barbeau commented on app/src/debug/google-services.json:
```diff
@@ -62,23 +62,23 @@
         "mobilesdk_app_id": "1:1022635885638:android:3ba5a7d13d995fc2",
         "android_client_info": {
-          "package_name": "com.simplecity.amp_pro"
+          "package_name": "edu.usf.sas.pal.muser"
```
Hmm, looks like the app already uses Firebase for something? We'll need to generate a new Google-services.json file for our app - I'll work on that. Could you check the Github Actions build failure to see what the problem is?

barbeau replied:
Also, please check the app and see what Firebase Google Services it currently seems to depend on, we'll need to switch those on.

BumbleFlash replied:
I checked the build logs and I think it's a syntax error and also certificate hash was missing from the JSON object. I have pushed a commit correcting the error and restoring the certificate to the request and it seems to have fixed the issue.

BumbleFlash committed:
changed app ID - (Bugs Fixed) - Syntax error and restored the certificate hash

BumbleFlash committed:
changed app ID - added the new google-services.json file.

barbeau commented on app/src/debug/google-services.json:
```diff
@@ -62,23 +62,23 @@
       "client_info": {
```
Hmmm...looks like there is more than one google-services.json file in the project? If so, I'm guessing we should delete the `debug` one, and just put ours in `main`. I'm guessing Shuttle has multiple Firebase accounts set up, one for debuggin and another for release. For our purposes, I think we only need one.

BumbleFlash replied:
Yes, you're right. One of them is for debugging. However, the devDebug build variant won't compile without the debug version of google-services.json

barbeau replied:
If you delete all google-services.json from the project, and then just put one file under the `app` directory, does that work?

See https://github.com/OneBusAway/onebusaway-android/tree/master/onebusaway-android for an example.

All the build variants should then inherit it.

BumbleFlash replied:
I tried doing that but the project failed to build without the google-services.json in the debug module.

barbeau replied:
Strange - that should work. The debug flavor failed to build? You may need to do "File->Invalidate caches/restart" to get all the build flavors to pick up the new config.

BumbleFlash replied:
I tried invalidating caches too, but it is still asking for the other json file in the debug.

barbeau replied:
Ok - I guess just duplicate the same file there and see if that works...

BumbleFlash replied:
Yes - It works when I duplicate the debug json file and place it in the app folder.

barbeau replied:
Ok - can you update this PR with that config that works?

BumbleFlash replied:
Sure. Will do.

BumbleFlash committed:
chore: moved google-services.json from debug to app.

BumbleFlash commented:
I don't understand why Github is failing to build. I ran it a few times in my system and it seems to build successfully.

BumbleFlash committed:
fix: uploaded google-services.json.

barbeau commented:
@BumbleFlash By default Android Studio will only build the flavor you have selected in the "Build variants" window, while the CI will build all debug variants. Possibly a variant is failing that's different than the one selected in Android Studio on your machine?

If you want to simulate the CI build on your machine, you can run this from the command-line, which will build all the debug variants:

```
./gradlew assembleDebug -x lint --stacktrace
```

barbeau commented:
Hmmm...looking at the variants for MUSER, though, there is only one debug variant, the `devDebug`?

barbeau commented:
For our release builds, we'll either want the `freeRelease` or `paidRelease` - so it's most important that those two along with `devDebug` build.

barbeau commented:
Looking at https://github.com/CUTR-at-USF/MUSER/runs/777335460, it says:

~~~
* What went wrong:
Execution failed for task ':app:processDevDebugGoogleServices'.
> File google-services.json is missing. The Google Services Plugin cannot function without it. 
   Searched Location: 
  /home/runner/work/MUSER/MUSER/app/src/dev/debug/google-services.json
  /home/runner/work/MUSER/MUSER/app/src/debug/dev/google-services.json
  /home/runner/work/MUSER/MUSER/app/src/dev/google-services.json
  /home/runner/work/MUSER/MUSER/app/src/debug/google-services.json
  /home/runner/work/MUSER/MUSER/app/src/devDebug/google-services.json
  /home/runner/work/MUSER/MUSER/app/google-services.json
~~~

Could you try deleting all `google-services.json` files again, and just put ours at `MUSER/app/google-services.json`, and push that to this PR and see if it builds?

BumbleFlash commented:
Sure, I can do that. However, I see that the Github has built the project after I added a commit that uploaded Debug's google-services.json. Do you still want me to upload our google-services.json to this PR?

barbeau commented:
Yes please. And to be clear, none of the google-services.json files from Shuttle should be left in this project - only ours.

BumbleFlash commented:
Sure! gotcha!

BumbleFlash committed:
chore: replaced google-services.json with ours.

BumbleFlash commented:
I guess it doesn't. Here's the error:-
`Execution failed for task ':app:processDevDebugGoogleServices'.`
`> No matching client found for package name 'edu.usf.sas.pal.muser.debug'`

barbeau commented:
Can you check the build.gradle to see if "debug" is being added to the applicationId somewhere?

BumbleFlash commented:
Yes, you're right. I removed the "debug" from build gradle and the project builds successfully.
Looking at the other build variants, the gradle file was referring to the whole new package "another.music.player". I have replaced that with our appID. However, it still needs the Keystore files to compile for the release variants. Do you want me to push this to the PR?

barbeau commented:
Yes, please, you can go ahead and push changes to PRs as you make them.

barbeau commented:
For signing release builds, let's use the same process that's used in GPSTest:
* Code - https://github.com/barbeau/gpstest/blob/master/GPSTest/build.gradle#L46
* Docs - https://github.com/barbeau/gpstest/blob/master/BUILD.MD#release-builds

Note that we should have the applicationID hard-coded in one place, and then the build.gradle may append something to it - but you shouldn't have to replace it in multiple places.

BumbleFlash committed:
fix: removed .debug

barbeau commented on app/build.gradle:
```diff
@@ -89,15 +89,14 @@ android {
 
         debug {
-            applicationIdSuffix ".debug"
+            applicationIdSuffix ""
```
You can just remove `applicationIdSuffix` entirely.

barbeau commented:
Per discussion today let's just use the existing config to read the keystore instead of using the version in GPSTest.

Here's the Google instructions for creating a keystore file:
https://developer.android.com/studio/publish/app-signing#generate-key

barbeau commented:
Per today's discussion we're also going to disable the older version of Crashlytics for release builds.

BumbleFlash committed:
chore: disabled crashlytics

barbeau commented:
Looks good, thanks @BumbleFlash!
